java/core/src/main/java/com/microsoft/bond/protocol/SimpleBinaryWriter.java

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+import com.microsoft.bond.BondEnum;
+import com.microsoft.bond.ProtocolType;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Implements Simple Binary serialization.
+ * Refer to {@see https://microsoft.github.io/bond/reference/cpp/simple__binary_8h_source.html} for details.
+ */
+public final class SimpleBinaryWriter implements ProtocolWriter {
+
+    // Indicates the protocol type when marshalling.
+    private static final ProtocolType MAGIC = ProtocolType.SIMPLE_PROTOCOL;
+
+    private final short protocolVersion;
+    private final BinaryStreamWriter writer;
+
+    public SimpleBinaryWriter(final OutputStream outputStream, final short protocolVersion) {
+        if (outputStream == null) {
+            throw new IllegalArgumentException("Argument stream must not be null");
+        }
+
+        if (protocolVersion != 1 && protocolVersion != 2) {
+            throw new IllegalArgumentException("Invalid protocol version: " + protocolVersion);
+        }
+
+        this.writer = new BinaryStreamWriter(outputStream);
+        this.protocolVersion = protocolVersion;
+    }
+
+    @Override
+    public boolean usesMarshaledBonded() {
+        return true;
+    }
+
+    @Override
+    public void writeVersion() throws IOException {
+        this.writer.writeInt16((short) MAGIC.value);
+        this.writer.writeInt16(this.protocolVersion);
+    }
+
+    @Override
+    public void writeFieldOmitted(
+            final BondDataType type, final int id, final FieldMetadata metadata) throws IOException {
+        assert !metadata.isDefaultNothing();
```

Comment:
> assert !metadata.isDefaultNothing(); [](start = 7, length = 37)

That's what I was saying this morning: an assert in library code. Are you going to remove it?